core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -114,43 +127,44 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
 
   /**
    * Returns the composition of this schedule and the specified schedule,
-   * by piping the output of this one into the input of the other, and summing
-   * delays produced by both.
+   * by piping the output of this one into the input of the other. Delays produced by this
+   * schedule will always run before the second schedule is evaluated.
    */
   final def >>>[R1 <: R, C](that: ZSchedule[R1, B, C]): ZSchedule[R1, A, C] =
     new ZSchedule[R1, A, C] {
       type State = (self.State, that.State)
       val initial = self.initial.zip(that.initial)
-      val update = (a: A, s: State) =>
-        self.update(a, s._1).flatMap { step1 =>
-          that.update(step1.finish(), s._2).map { step2 =>
-            step1.combineWith(step2)(_ && _, _ + _).rightMap(_._2)
-          }
-        }
+      val extract = (a: A, s: (self.State, that.State)) => that.extract(self.extract(a, s._1), s._2)
+
+      val update = (a: A, s: (self.State, that.State)) =>
+        for {
+          s1 <- self.update(a, s._1)
+          s2 <- that.update(self.extract(a, s._1), s._2)
+        } yield (s1, s2)
     }
 
   /**
    * Returns a new schedule that continues as long as either schedule continues,
    * using the minimum of the delays of the two schedules.
    */
   final def ||[R1 <: R, A1 <: A, C](that: ZSchedule[R1, A1, C]): ZSchedule[R1, A1, (B, C)] =
-    combineWith(that)(_ || _, _ min _)
+    new ZSchedule[R1, A1, (B, C)] {
+      type State = (self.State, that.State)
+      val initial = self.initial zip that.initial
```

Comment:
Nice and clean!

We may want to consider a test for [this one](https://github.com/zio/zio/issues/1909) so we can close it out in this PR.

Reply:
Good point! added

Reply:
I'll also update how the tests are grouped/organized in a seperate pr